ddcommon-net/src/http.rs

```diff
@@ -0,0 +1,287 @@
+        return Err(Error::Io(io::Error::new(
+            io::ErrorKind::InvalidInput,
+            "URI scheme must be http",
+        )));
+    }
+
+    let authority = request
+        .uri()
+        .authority()
+        .ok_or(Error::Io(io::Error::new(
+            io::ErrorKind::InvalidInput,
+            "URI must have host",
+        )))?
+        .as_str();
+    let stream = TcpStream::connect(authority).await?;
+    let hyper_wrapper = TokioIo::new(stream);
+
+    Ok(send_via_io(request, hyper_wrapper).await?)
+}
+
+pub async fn send_https<B>(request: Request<B>) -> StdResult<Response<Incoming>, Error>
+where
+    B: Body + Send + 'static,
+    B::Data: Send,
+    B::Error: Into<Box<dyn std::error::Error + Send + Sync>>,
+{
+    let uri = request.uri();
+
+    // This _should_ be a redundant check, caller should only call this if
+    // they expect it's an https connection to begin with.
+    if uri.scheme_str() != Some("https") {
+        return Err(Error::Io(io::Error::new(
+            io::ErrorKind::InvalidInput,
+            "URI scheme must be https",
+        )));
+    }
+
+    let server_name = ServerName::try_from(uri.to_string())?;
+    let connector = {
+        let config = rustls::ClientConfig::builder()
+            .with_root_certificates(rustls::RootCertStore::empty())
+            .with_no_client_auth();
+        tokio_rustls::TlsConnector::from(sync::Arc::new(config))
+    };
+
+    let tcp_stream = {
+        let authority = uri.authority().ok_or(Error::Io(io::Error::new(
+            io::ErrorKind::InvalidInput,
+            "URI must have a host",
+        )))?;
+        TcpStream::connect(authority.as_str()).await?
+    };
+
+    let stream = connector.connect(server_name, tcp_stream).await?;
+    let hyper_wrapper = TokioIo::new(stream);
+    Ok(send_via_io(request, hyper_wrapper).await?)
+}
+
+async fn send_via_io<T, B>(
+    request: Request<B>,
+    io: T,
+) -> StdResult<Response<Incoming>, hyper::Error>
+where
+    T: Read + Write + Send + Unpin + 'static,
+    B: Body + Send + 'static,
+    B::Data: Send,
+    B::Error: Into<Box<dyn std::error::Error + Send + Sync>>,
+{
+    let (mut sender, connection) = hyper::client::conn::http1::handshake(io).await?;
+
+    // The docs say we need to poll this to drive it to completion, but they
+    // never directly use the return type or anything:
+    // https://hyper.rs/guides/1/client/basic/
+    let _todo = tokio::spawn(async move { connection.await });
+
+    sender.send_request(request).await
+}
+
+pub fn parse_path_from_uri(uri: &Uri) -> io::Result<path::PathBuf> {
+    // This _should_ be a redundant check, caller should only call this if
+    // they expect it's a unix domain socket or windows named pipe.
+    if uri.scheme_str() != Some("unix") || uri.scheme_str() != Some("windows") {
+        return Err(io::Error::new(
+            io::ErrorKind::InvalidInput,
+            "URI scheme must be unix or windows",
```

Comment:
```suggestion
            format!("URI scheme must be unix or windows: was {}", uri.scheme_str()),
```

Reply:
Since `scheme_str` is actually `Option<&str>`, my commit is a bit different but I improved these, thanks!